pkg/printers/humanreadable.go

```diff
@@ -95,26 +104,90 @@ func (h *HumanReadablePrinter) EnsurePrintHeaders() {
 // See validatePrintHandlerFunc for required method signature.
 func (h *HumanReadablePrinter) Handler(columns, columnsWithWide []string, printFunc interface{}) error {
 	printFuncValue := reflect.ValueOf(printFunc)
-	if err := h.validatePrintHandlerFunc(printFuncValue); err != nil {
-		glog.Errorf("Unable to add print handler: %v", err)
-		return err
+	entry := &handlerEntry{
+		columns:         columns,
+		columnsWithWide: columnsWithWide,
+	}
+	if err := ValidateRowPrintHandlerFunc(printFuncValue); err == nil {
+		entry.rowPrintFunc = printFuncValue
+	} else {
+		if nestedErr := ValidatePrintHandlerFunc(printFuncValue); nestedErr == nil {
+			entry.printFunc = printFuncValue
+		} else {
+			utilruntime.HandleError(fmt.Errorf("unable to register print function: %v %v", err, nestedErr))
+			return err
+		}
 	}
 
 	objType := printFuncValue.Type().In(0)
-	h.handlerMap[objType] = &handlerEntry{
+	h.handlerMap[objType] = entry
+	return nil
+}
+
+// TableHandler adds a print handler with a given set of columns to HumanReadablePrinter instance.
+// See validatePrintHandlerFunc for required method signature.
+func (h *HumanReadablePrinter) TableHandler(columnDefinitions []metav1alpha1.TableColumnDefinition, printFunc interface{}) error {
+	printFuncValue := reflect.ValueOf(printFunc)
+	var columns []string
+	var columnsWithWide []string
+	for _, column := range columnDefinitions {
+		if column.Priority == 0 {
+			columns = append(columns, strings.ToUpper(column.Name))
+		} else {
+			columnsWithWide = append(columnsWithWide, strings.ToUpper(column.Name))
+		}
+	}
+	entry := &handlerEntry{
 		columns:         columns,
 		columnsWithWide: columnsWithWide,
-		printFunc:       printFuncValue,
+	}
+	if err := ValidateRowPrintHandlerFunc(printFuncValue); err == nil {
+		entry.rowPrintFunc = printFuncValue
+	} else {
+		if nestedErr := ValidatePrintHandlerFunc(printFuncValue); nestedErr == nil {
+			entry.printFunc = printFuncValue
+		} else {
+			utilruntime.HandleError(fmt.Errorf("unable to register print function: %v %v", err, nestedErr))
+			return err
+		}
+	}
+
+	objType := printFuncValue.Type().In(0)
+	h.handlerMap[objType] = entry
```

Comment:
I would expect an error or panic on overwriting this value to help coders find problems early.